addons/website_event_sale/tests/test_website_event_sale_cart.py

```diff
@@ -0,0 +1,30 @@
+# coding: utf-8
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo.addons.website.tools import MockRequest
+from odoo.tests.common import TransactionCase, tagged
+
+
+@tagged('post_install', '-at_install')
+class WebsiteEventSaleCart(TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.website = cls.env['website'].browse(1)
+        cls.WebsiteSaleController = WebsiteSale()
+        cls.public_user = cls.env.ref('base.public_user')
+        cls.product = cls.env.ref("event_sale.product_product_event")
+        cls.product.website_published = True
+        ended_event = cls.env.ref("event.event_4")
+        cls.ongoing_event = cls.env.ref("event.event_7")
+        # get rid of other events so we can make a resilient test
+        cls.product.event_ticket_ids.filtered(lambda x: x.event_id not in (ended_event + cls.ongoing_event))
+
+    def test_event_product_add_to_cart(self):
+        website = self.website.with_user(self.public_user)
+        product = self.product
+        with MockRequest(product.with_user(self.public_user).env, website=self.website.with_user(self.public_user)):
+            self.WebsiteSaleController.cart_update_json(product_id=product.id, add_qty=1)
+            line = website.sale_get_order().order_line
+            self.assertEqual(line.event_id, self.ongoing_event)
```

Comment:
Overall, I didn't see any class that does any meaningful set-up to help with this case so I did some set up based on the demo data you used :)

Omitting details to hopefully have the minimal set up that still makes sense for the case.

```py
from datetime import datetime, timedelta
from freezegun import freeze_time

from odoo.addons.mail.tests.common import mail_new_test_user
from odoo.addons.website_sale.controllers.main import WebsiteSale
from odoo.addons.website.tools import MockRequest
from odoo.tests.common import TransactionCase, tagged, users

@tagged('post_install', '-at_install')
class WebsiteEventSaleCart(TransactionCase):

    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        cls.today = datetime(2039, 4, 29, 8, 0)
        cls.website = cls.env['website'].get_current_website()
        cls.WebsiteSaleController = WebsiteSale()

        cls.public_user = mail_new_test_user(
            cls.env,
            name='Public user',
            login='public_user_website_event_sale_cart',
            email='public_user@example.com',
            groups='base.group_public',
        )
        cls.product = cls.env['product.product'].create({
            'detailed_type': 'event',
            'invoice_policy': 'order',
            'name': 'Event Registration',
            'website_published': True,
        })

        cls.ongoing_stage = cls.env['event.stage'].create({
            "name": "Ongoing",
            "description": "Ongoing",
            "sequence": 0,
            "pipe_end": False,
        })
        cls.end_stage = cls.env['event.stage'].create({
            "name": "Ended",
            "description": "Fully ended",
            "sequence": 1,
            "pipe_end": True,
        })

        cls.ended_event = cls.env['event.event'].create({
            "name": "Business workshops",
            "date_begin": cls.today.replace(hour=18) - timedelta(days=5),
            "date_end": cls.today.replace(hour=22) - timedelta(days=5),
            "stage_id": cls.end_stage.id,
            "kanban_state": "done",
        })
        cls.env['event.event.ticket'].create({
            "name": "General Admission",
            "event_id": cls.ended_event.id,
            "end_sale_datetime": cls.today - timedelta(days=30),
            "product_id": cls.product.id,
        })

        cls.ongoing_event = cls.env['event.event'].create({
            "name": "OpenWood Collection Online Reveal",
            "stage_id": cls.ongoing_stage.id,
            "auto_confirm": True,
            "date_begin": cls.today.replace(hour=5) - timedelta(days=1),
            "date_end": cls.today.replace(hour=15) + timedelta(days=1)
        })
        cls.env['event.event.ticket'].create({
            "name": "Standard",
            "event_id": cls.ongoing_event.id,
            "end_sale_datetime": cls.today.replace(hour=15) + timedelta(days=2),
            "product_id": cls.product.id,
        })

    @users('public_user_website_event_sale_cart')
    def test_event_product_add_to_cart(self):
        with freeze_time(self.today), MockRequest(self.env, website=self.website.with_user(self.env.user)):
            self.WebsiteSaleController.cart_update_json(product_id=self.product.id, add_qty=1)
            line = self.website.sale_get_order().order_line
            self.assertEqual(line.event_id.name, self.ongoing_event.name)

```